### ISSUE TITLE
Critical error: PointAnnotation click causes an app crash on Android platform

### ISSUE THREAD

**Issue by blisterrabbithutch** [rnmapbox/maps#1475]
Hi. I've started to develop an app with Mapbox integration and I caught a problem which makes the further development impossible. I've tried to start my RN application on Android Emulator and on my physical device (Mi 9T)

**Describe the bug**
after Mapbox installation, on Android (iOS is working well as expected without any bugs and errors) when I added inside my MapView -> PointAnnotation marker (as in repo's example) I saw that press on any marker causes an app's crash with terminal error.

Also, I have a warning in my terminal when mapbox screen loads ->
**Screenshot of a warning:**
![image](https://user-images.githubusercontent.com/27296220/127548010-d2b06255-35c6-43b1-9bce-65cb90598346.png)
https://monosnap.com/file/YIefg0xJGl4qA1hOlACa1WVgP6GvVa

BTW, How can I resolve this warning?

**My build:**
npx react-native init myProject
...then package.json libraries (react-native-snap-carousel, react-navigation (and all dependencies), native-base, react-native-svg
npm install @react-native-mapbox-gl/maps --save
after that I completed all steps from this page -> https://github.com/react-native-mapbox-gl/maps/blob/master/android/install.md
npm start (npx react-native start) -> npx react-native run-android

After that, my app builds and launches successfully on my phone/emulator.

**My package.json**
```
"dependencies": {
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-mapbox-gl/maps": "^8.2.1",
    "@react-navigation/native": "^5.9.4",
    "@react-navigation/stack": "^5.14.5",
    "native-base": "^3.0.6",
    "react": "17.0.1",
    "react-native": "0.64.2",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-reanimated": "^2.2.0",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.4.0",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "^12.1.1",
    "styled-components": "^5.3.0",
    "styled-system": "^5.1.5"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-jest": "^26.6.3",
    "eslint": "7.14.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.64.0",
    "react-test-renderer": "17.0.1"
  },
```

**Error message -**
```
 JavaScript logs will appear in your browser console
Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:592:3)
    at tryReadSync (fs.js:366:20)
    at Object.readFileSync (fs.js:403:19)
    at UnableToResolveError.buildCodeFrameMessage (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:347:17)
    at new UnableToResolveError (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:333:35)
    at ModuleResolver.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:211:15)
    at DependencyGraph.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph.js:413:43)
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/lib/transformHelpers.js:317:42
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/Server.js:1471:14
    at Generator.next (<anonymous>)
Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:592:3)
    at tryReadSync (fs.js:366:20)
    at Object.readFileSync (fs.js:403:19)
    at UnableToResolveError.buildCodeFrameMessage (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:347:17)
    at new UnableToResolveError (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:333:35)
    at ModuleResolver.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:211:15)
    at DependencyGraph.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph.js:413:43)
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/lib/transformHelpers.js:317:42
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/Server.js:1471:14
    at Generator.next (<anonymous>)
 MAP  ./index.js 

 MAP  ./index.js ░░░░░░░░░░░░░░░░ 0.0% (0/1)Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:592:3)
    at tryReadSync (fs.js:366:20)
    at Object.readFileSync (fs.js:403:19)
    at UnableToResolveError.buildCodeFrameMessage (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:347:17)
    at new UnableToResolveError (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:333:35)
    at ModuleResolver.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:211:15)
    at DependencyGraph.resolveDependency (/Users/macbook/Projects/palazzo/node_modules/metro/src/node-haste/DependencyGraph.js:413:43)
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/lib/transformHelpers.js:317:42
    at /Users/macbook/Projects/palazzo/node_modules/metro/src/Server.js:1471:14
    at Generator.next (<anonymous>)
```

**Terminal screenshot of a problem**
![image](https://user-images.githubusercontent.com/27296220/127619077-125f0bf9-dbde-4a7f-93c0-80e632523062.png)
![image](https://user-images.githubusercontent.com/27296220/127548095-a528863f-3308-4971-bfe0-a0b6e5397ce7.png)
![image](https://user-images.githubusercontent.com/27296220/127548107-f6a4580f-7e33-4b5b-baf3-de2618be8ce1.png)

https://monosnap.com/file/bfFvj3NCMXw84BQJsxSEYr3peA9zcu
https://monosnap.com/file/Z6LWRYZNKVDLvqYmMAaCkz076cAn9i

**Screenshot of Mapbox screen**
![image](https://user-images.githubusercontent.com/27296220/127547866-f7a9491a-3aae-4c33-ae95-963009dc9f48.png)
https://monosnap.com/file/cwEH7BAOzvCGRDSYA0cYHlWg0sFdUN

**Code Example:**

```
import React from 'react';
import {Dimensions, ScrollView, StyleSheet, View} from 'react-native';
import MapboxGL from "@react-native-mapbox-gl/maps";

MapboxGL.setAccessToken("<MyDefaultPublicToken>");
MapboxGL.setConnected(true);

export const MapPage = () => {
  return (
    <View style={styles.container}>
      <ScrollView style={styles.pageContent}>
        <MapboxGL.MapView
          style={styles.map}
          styleURL="{MyStyleUrl}"
        >
          <MapboxGL.Camera
            animationDuration={0}
            zoomLevel={12}
            centerCoordinate={[
              37.030866,
              55.778953
            ]}
          />
          <MapboxGL.PointAnnotation
            key="pointAnnotation"
            id="pointAnnotation"
            coordinate={[
              37.030866,
              55.778953
            ]}
          />
        </MapboxGL.MapView>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  pageContent: {
    flex: 1,
  },
  map: {
    height: Dimensions.get("window").height,
  },
});

```

End of Code Example.

**Expected behavior**
Click on marker shouldn't lead to app's crash

**Versions:**

- Platform: Android
- Platform OS: [emulator Pixel 4 XL Android 11.0, (API 30), Mi 9T Android 9.0]
- Emulator/Real device: both
- Dev OS: [OSX 10.15.4]
- react-native-mapbox-gl Version [^8.2.1]
- React Native Version [0.64.2]

Also, I don't know why are there another mapbox dependencies in example package.json ? Are they neccessary? If so - why there is no information about their installation.
https://github.com/react-native-mapbox-gl/maps/blob/master/example/package.json

android/app/build.gradle
```
    implementation 'com.mapbox.mapboxsdk:mapbox-android-telemetry:6.1.0'
    implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:9.6.1'
    implementation 'com.mapbox.mapboxsdk:mapbox-sdk-services:5.6.0'
    implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.9.0'
```

Gradle - 6.7

**Comment by ferdicus** [comment]
I've used the android dependencies for mapbox that you mentioned:
```
    implementation 'com.mapbox.mapboxsdk:mapbox-android-telemetry:6.1.0'
    implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:9.6.1'
    implementation 'com.mapbox.mapboxsdk:mapbox-sdk-services:5.6.0'
    implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.9.0'
```

and I can confirm this happening when using:
`implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.9.0'`
as a workaround, try with 
`    implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.8.0'`
instead, 
which didn't crash in my tests on physical device with the example app.

<br>

Not quite sure what the issue is yet, here the callstack though (in textform):
```sh
Process: com.rnmapboxglexample, PID: 23180
java.lang.AbstractMethodError: abstract method "boolean com.mapbox.mapboxsdk.plugins.annotation.OnAnnotationClickListener.onAnnotationClick(com.mapbox.mapboxsdk.plugins.annotation.Annotation)"
	at com.mapbox.mapboxsdk.plugins.annotation.AnnotationManager$MapClickResolver.onMapClick(AnnotationManager.java:381)
	at com.mapbox.mapboxsdk.maps.MapGestureDetector.notifyOnMapClickListeners(MapGestureDetector.java:996)
	at com.mapbox.mapboxsdk.maps.MapGestureDetector$StandardGestureListener.onSingleTapConfirmed(MapGestureDetector.java:359)
	at com.mapbox.android.gestures.StandardGestureDetector$1.onSingleTapConfirmed(StandardGestureDetector.java:83)
	at android.view.GestureDetector$GestureHandler.handleMessage(GestureDetector.java:323)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:241)
	at android.app.ActivityThread.main(ActivityThread.java:7582)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)
```

<br>

-----

<br>

Concerning the `GlyphsRasterizationMode` warning, this is due to an issue mentioned in this [comment](https://github.com/react-native-mapbox-gl/maps/issues/1211#issuecomment-841817849). 
**TLDR**: It's annoying but doesn't hurt. 

If you want to get rid of it, I'd recommend suppressing it in `Logger` like so:

```ts
import { Logger } from '@react-native-mapbox-gl/maps';

// edit logging messages
Logger.setLogCallback(log => {
  const { message } = log;
  if (
    message.match('Falling back to MGLIdeographsRasterizedLocally') ||
    message.match('MapRenderer::onSurfaceCreated GlyphsRasterizationMode')
  ) {
    return true;
  }
  return false;
});
```
Also see:
https://github.com/mapbox/mapbox-gl-native/issues/15341#issuecomment-522889062
https://github.com/mapbox/mapbox-gl-native-ios/issues/589

<br>

-----

<br>

Concerning `/example` app and other mapbox dependencies inside the `package.json`. 
Do you mean `@mapbox/mapbox-sdk`?? 
That's just to get the directions for one example screen - has nothing to do with dependencies to get the lib running on your app. 

<br>

-----

<br>

One last thing - please don't use screenshots of log output - it makes it really hard to search for. 


Cheers 🍻
